PHPCompatibility/Tests/BaseSniffTestCase.php

```diff
@@ -377,4 +377,45 @@ private function gatherIssues($issuesArray)
 
         return $allIssues;
     }
+
+    /**
+     * Run assertions that the fixer output matches the ".fixed" file for a particular sniff.
+     *
+     * @since 10.0.0
+     *
+     * @param string $pathToTestFile   Absolute path to the file to sniff.
+     *                                 Allows for passing __FILE__ from the unit test
+     *                                 file. In that case, the test case file is presumed
+     *                                 to have the same name, but with an `inc` extension.
+     * @param string $targetPhpVersion Value of 'testVersion' to set on PHPCS object.
+     *
+     * @return void
+     */
+    protected function assertFixerOutputMatches($pathToTestFile, $targetPhpVersion = 'none')
+    {
+        $pathToFixedFile = $pathToTestFile . '.fixed';
+
+        $this->assertTrue(file_exists($pathToTestFile), 'Test file exists');
+        $this->assertTrue(file_exists($pathToFixedFile), 'Fixed file exists');
+
+        $phpcsFile = $this->sniffFile($pathToTestFile, $targetPhpVersion);
+
+        $this->assertGreaterThan(0, $phpcsFile->getFixableCount(), 'There are fixable errors or warnings');
+
+        // Attempt to fix the errors.
+        $phpcsFile->fixer->fixFile();
```

Comment:
The test run order is not fixed. If the fixer test would be run (for whatever reason) before the "sniffing tests", the sniff tests would likely fail as the `$phpcsFile` object will have been modified by the fixer run.